command/agent/config.go

```diff
@@ -217,6 +217,9 @@ type ClientConfig struct {
 	// NoHostUUID disables using the host's UUID and will force generation of a
 	// random UUID.
 	NoHostUUID *bool `mapstructure:"no_host_uuid"`
+
+	// ServerJoin contains information that is used to attempt to join servers
+	ServerJoin *ServerJoin `mapstructure:"server_join_info"`
```

Comment:
why `_info`?

Reply:
+1, would suggest calling this server_join or server_join_config maybe?

Reply:
This is left over from a rename, I'll move to just `server_join`